File: Assets/Float.cs
using UnityEngine;
using System.Collections;

public class Float : MonoBehaviour {
	
	private bool floatup;
	
	private Vector3 gravity;

	public GameObject obiect;
	// Use this for initialization
	void Start () {
		gravity = new Vector3 (0,0,0);
		Physics.gravity = gravity;
		obiect = this.gameObject;
		
	}
	void FixedUpdate () {
	    if(Input.GetKeyDown(KeyCode.W)) {
	        floatingup(8);
			}
			if (Input.GetKeyDown(KeyCode.S)) {
	        floatingdown(8);
			}
			 if (Input.GetKeyDown(KeyCode.A)) {
	        floatingleft(8);
			}
			 if(Input.GetKeyDown(KeyCode.D)) {
	        floatingright(8);
			}
	}
	public void floatingup(float i){	
    obiect.rigidbody.AddForce(0,i,0);
	}
	public void floatingdown(float i){
		obiect.rigidbody.AddForce(0,-i,0);
	}
	public void floatingleft(float i){
		obiect.rigidbody.AddForce(-i,0,0);
	}
	public void floatingright(float i){
		obiect.rigidbody.AddForce(i,0,0);
	}
	
	void OnCollisionEnter(Collision c){
		Debug.Log("im " + obiect.name + " and i collided with " + c.gameObject.name);
		
	}
	
	public void Xtremes(){
		
	}
}

File: Assets/Scripts/Float.js
var floatup;
var stopVelocity;
var rigBody;
public var object;

function Start(){
Physics.gravity = Vector3(0, -0.05, 0);
   	
    stopVelocity = Vector3(0,0,0);
    rigBody = rigidbody;
}
function Update(){
	//Debug.Log("-" + rigidbody.velocity );
	//rigidbody.velocity = constantVelocity;
	Debug.Log(floatup);
    if(Input.GetKeyDown(KeyCode.A)) {
        floatingup();
        floatup = true;
        } else {
       		 floatup = false;
        }
}
function floatingup(){
    rigidbody.AddForce(0,100,0);
    
    
    if (floatup == false){
    	yield WaitForSeconds(2);
    	//rigidbody.velocity = stopVelocity;
    }
    
    
}